File: custconnect_mobile/lib/presentation/screens/dashboard_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:lucide_icons/lucide_icons.dart';
import 'package:go_router/go_router.dart';
import '../providers/auth_provider.dart';

class DashboardScreen extends ConsumerWidget {
  const DashboardScreen({Key? key}) : super(key: key);

  void _logout(BuildContext context, WidgetRef ref) async {
    await ref.read(authStateProvider.notifier).logout();
    if (context.mounted) {
      context.go('/welcome');
    }
  }

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final theme = Theme.of(context);
    final user = ref.watch(authStateProvider).value;
    final initials = user != null ? '${user.firstName[0]}${user.lastName[0]}'.toUpperCase() : 'AJ';

    return Scaffold(
      body: SafeArea(
        child: Stack(
          children: [
            SingleChildScrollView(
              padding: const EdgeInsets.fromLTRB(20, 20, 20, 120),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.stretch,
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text('Good Morning,', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w600, color: theme.colorScheme.secondary)),
                          Text(user?.firstName ?? 'Student', style: const TextStyle(fontSize: 28, fontWeight: FontWeight.w900, letterSpacing: -0.5)),
                        ],
                      ),
                      Row(
                        children: [
                          Container(
                            padding: const EdgeInsets.all(10),
                            decoration: BoxDecoration(color: theme.colorScheme.surface, borderRadius: BorderRadius.circular(14)),
                            child: Icon(LucideIcons.bell, color: theme.colorScheme.secondary),
                          ),
                          const SizedBox(width: 15),
                          GestureDetector(
                            onTap: () => _logout(context, ref),
                            child: Container(
                              width: 48,
                              height: 48,
                              decoration: BoxDecoration(
                                color: theme.primaryColor,
                                borderRadius: BorderRadius.circular(16),
                                border: Border.all(color: Colors.white, width: 3),
                              ),
                              child: Center(
                                child: Text(initials, style: const TextStyle(color: Colors.white, fontWeight: FontWeight.w900, fontSize: 16)),
                              ),
                            ),
                          )
                        ],
                      )
                    ],
                  ),
                  const SizedBox(height: 30),
                  // CGPA Banner
                  Container(
                    padding: const EdgeInsets.all(30),
                    decoration: BoxDecoration(
                      color: theme.colorScheme.secondary,
                      borderRadius: BorderRadius.circular(35),
                    ),
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Container(
                              padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                              decoration: BoxDecoration(color: Colors.white.withOpacity(0.15), borderRadius: BorderRadius.circular(12)),
                              child: Row(
                                children: const [
                                  Icon(LucideIcons.trendingUp, size: 12, color: Colors.white),
                                  SizedBox(width: 6),
                                  Text("DEAN'S LIST", style: TextStyle(color: Colors.white, fontSize: 10, fontWeight: FontWeight.w900, letterSpacing: 1)),
                                ],
                              ),
                            ),
                            const Text('3.85', style: TextStyle(color: Colors.white, fontSize: 50, fontWeight: FontWeight.w900)),
                            Text('Current Cumulative GPA', style: TextStyle(color: Colors.white.withOpacity(0.5), fontSize: 12, fontWeight: FontWeight.w600)),
                          ],
                        ),
                        Container(
                          width: 70, height: 70,
                          decoration: BoxDecoration(
                            shape: BoxShape.circle,
                            border: Border.all(color: Colors.white.withOpacity(0.1), width: 6),
                          ),
                          child: const Center(child: Text('92%', style: TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.w900))),
                        )
                      ],
                    ),
                  ),
                  const SizedBox(height: 35),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      const Text('Quick Actions', style: TextStyle(fontSize: 20, fontWeight: FontWeight.w900)),
                      Text('Customize', style: TextStyle(color: theme.primaryColor, fontSize: 14, fontWeight: FontWeight.bold)),
                    ],
                  ),
                  const SizedBox(height: 20),
                  Row(
                    children: [
                      Expanded(child: _ActionCard(title: 'GPA Calc', desc: 'Track Performance', iconData: LucideIcons.barChart3, color: theme.primaryColor, onTap: ()=>context.push('/gpa'))),
                      const SizedBox(width: 15),
                      Expanded(child: _ActionCard(title: 'Library', desc: 'Digital Assets', iconData: LucideIcons.bookOpen, color: theme.colorScheme.secondary, onTap: ()=>context.push('/resources'))),
                    ],
                  ),
                  const SizedBox(height: 15),
                  Row(
                    children: [
                      Expanded(child: _ActionCard(title: 'Social', desc: 'Campus Feed', iconData: LucideIcons.messageSquare, color: Colors.amber.shade700, onTap: (){})),
                      const SizedBox(width: 15),
                      Expanded(child: _ActionCard(title: 'Transport', desc: 'Live Tracking', iconData: LucideIcons.bus, color: Colors.orange.shade700, onTap: (){})),
                    ],
                  ),
                  const SizedBox(height: 35),
                  const Text('Academic Status', style: TextStyle(fontSize: 20, fontWeight: FontWeight.w900)),
                  const SizedBox(height: 15),
                  Container(
                    padding: const EdgeInsets.all(20),
                    decoration: BoxDecoration(color: theme.colorScheme.surface, borderRadius: BorderRadius.circular(24)),
                    child: Row(
                      children: [
                        Container(
                          width: 44, height: 44,
                          decoration: BoxDecoration(color: Colors.amber.shade50, borderRadius: BorderRadius.circular(14)),
                          child: Icon(LucideIcons.bell, color: Colors.brown.shade700, size: 20),
                        ),
                        const SizedBox(width: 15),
                        Expanded(
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: const [
                              Text('Midterm schedules are now available for Spring 2026.', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500)),
                              SizedBox(height: 4),
                              Text('Posted 2h ago', style: TextStyle(fontSize: 11, fontWeight: FontWeight.bold, color: Colors.grey)),
                            ],
                          ),
                        )
                      ],
                    ),
                  )
                ],
              ),
            ),
            
            // Bottom Tab Bar matching Figma design
            Align(
              alignment: Alignment.bottomCenter,
              child: Container(
                margin: const EdgeInsets.only(bottom: 20, left: 20, right: 20),
                height: 85,
                decoration: BoxDecoration(
                  color: theme.colorScheme.surface,
                  borderRadius: BorderRadius.circular(30),
                  boxShadow: [BoxShadow(color: Colors.black12, blurRadius: 20, offset: const Offset(0, 10))],
                ),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceAround,
                  children: [
                    Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        Icon(LucideIcons.home, color: theme.primaryColor),
                        const SizedBox(height: 6),
                        Container(width: 4, height: 4, decoration: BoxDecoration(color: theme.primaryColor, shape: BoxShape.circle))
                      ],
                    ),
                    IconButton(icon: const Icon(LucideIcons.bookOpen, color: Colors.grey), onPressed: ()=>context.push('/resources')),
                    Container(
                      transform: Matrix4.translationValues(0, -25, 0),
                      width: 64, height: 64,
                      decoration: BoxDecoration(
                        color: theme.colorScheme.secondary,
                        shape: BoxShape.circle,
                        border: Border.all(color: theme.scaffoldBackgroundColor, width: 5),
                      ),
                      child: const Icon(LucideIcons.plus, color: Colors.white, size: 30),
                    ),
                    IconButton(icon: const Icon(LucideIcons.barChart3, color: Colors.grey), onPressed: ()=>context.push('/gpa')),
                    IconButton(icon: const Icon(LucideIcons.user, color: Colors.grey), onPressed: ()=>_logout(context, ref)),
                  ],
                ),
              ),
            )
          ],
        ),
      ),
    );
  }
}

class _ActionCard extends StatelessWidget {
  final String title;
  final String desc;
  final IconData iconData;
  final Color color;
  final VoidCallback onTap;

  const _ActionCard({
    required this.title, required this.desc, required this.iconData, required this.color, required this.onTap
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        decoration: BoxDecoration(
          color: Theme.of(context).colorScheme.surface,
          borderRadius: BorderRadius.circular(24),
          border: Border(top: BorderSide(color: color, width: 3)),
          boxShadow: [BoxShadow(color: Colors.black12, blurRadius: 5)],
        ),
        padding: const EdgeInsets.all(20),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Container(
              width: 48, height: 48,
              decoration: BoxDecoration(color: color.withOpacity(0.15), borderRadius: BorderRadius.circular(16)),
              child: Icon(iconData, color: color),
            ),
            const SizedBox(height: 15),
            Text(title, style: const TextStyle(fontWeight: FontWeight.w800, fontSize: 16)),
            const SizedBox(height: 4),
            Text(desc, style: const TextStyle(fontWeight: FontWeight.w500, fontSize: 11, color: Colors.grey)),
          ],
        ),
      ),
    );
  }
}

File: android_legacy_backup/src/screens/DashboardScreen.js
import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    ScrollView,
    TouchableOpacity,
    SafeAreaView,
    Dimensions,
    Platform
} from 'react-native';
import { useTheme } from '../contexts/ThemeContext';
import {
    Home,
    BookOpen,
    Plus,
    BarChart3,
    User,
    MessageSquare,
    Bus,
    Bell,
    TrendingUp,
    Moon,
    Sun
} from 'lucide-react-native';

const { width } = Dimensions.get('window');

const ActionCard = ({ title, icon, color, description, onPress, colors, shadowStyle }) => (
    <TouchableOpacity
        style={[styles.card, { borderTopColor: color, borderTopWidth: 3, backgroundColor: colors.surface }, shadowStyle]}
        onPress={onPress}
        activeOpacity={0.7}
    >
        <View style={styles.cardHeaderRow}>
            <View style={[styles.iconContainer, { backgroundColor: color + '15' }]}>
                {icon}
            </View>
            <View style={styles.cardContent}>
                <Text style={[styles.cardTitle, { color: colors.text.primary }]}>{title}</Text>
                <Text style={[styles.cardDescription, { color: colors.text.secondary }]}>{description}</Text>
            </View>
        </View>
    </TouchableOpacity>
);

export default function DashboardScreen({ user, onNavigate, onLogout }) {
    const { colors, isDarkMode, toggleTheme, shadows, spacing } = useTheme();
    const initials = user ? `${user.firstName?.[0] || ''}${user.lastName?.[0] || ''}` : 'AJ';

    return (
        <SafeAreaView style={[styles.container, { backgroundColor: colors.background }]}>
            <ScrollView
                showsVerticalScrollIndicator={false}
                contentContainerStyle={[styles.scrollContent, { padding: spacing.lg }]}
            >
                <View style={styles.header}>
                    <View>
                        <Text style={[styles.greeting, { color: colors.text.secondary }]}>Good Morning,</Text>
                        <Text style={[styles.userName, { color: colors.text.primary }]}>{user?.firstName || 'Student'}</Text>
                    </View>
                    <View style={styles.headerRight}>
                        <TouchableOpacity
                            style={[styles.iconButton, { backgroundColor: colors.surface }, shadows.sm]}
                        >
                            <Bell size={24} color={colors.secondary} />
                        </TouchableOpacity>
                        <TouchableOpacity
                            style={[styles.profileAvatar, { backgroundColor: colors.primary }]}
                            onPress={onLogout}
                        >
                            <Text style={styles.avatarText}>{initials}</Text>
                        </TouchableOpacity>
                    </View>
                </View>

                {/* CGPA Banner */}
                <View style={[styles.banner, { backgroundColor: isDarkMode ? colors.surface : colors.secondary }, shadows.lg]}>
                    <View style={styles.bannerLeft}>
                        <View style={styles.standingBadge}>
                            <TrendingUp size={12} color="white" />
                            <Text style={styles.standingText}>DEAN'S LIST</Text>
                        </View>
                        <Text style={styles.bannerValue}>3.85</Text>
                        <Text style={[styles.bannerLabel, { color: isDarkMode ? colors.text.secondary : 'rgba(255,255,255,0.5)' }]}>Current Cumulative GPA</Text>
                    </View>
                    <View style={styles.bannerRight}>
                        <View style={[styles.progressCircle, { borderTopColor: colors.primaryLight }]}>
                            <Text style={styles.progressText}>92%</Text>
                        </View>
                    </View>
                </View>

                <View style={styles.sectionHeader}>
                    <Text style={[styles.sectionTitle, { color: colors.text.primary }]}>Quick Actions</Text>
                    <TouchableOpacity>
                        <Text style={[styles.viewAllText, { color: colors.primary }]}>Customize</Text>
                    </TouchableOpacity>
                </View>

                <View style={styles.grid}>
                    <View style={styles.gridRow}>
                        <ActionCard
                            title="GPA Calc"
                            icon={<BarChart3 size={22} color={colors.primary} />}
                            color={colors.primary}
                            description="Track Performance"
                            onPress={() => onNavigate('gpa')}
                            colors={colors}
                            shadowStyle={shadows.sm}
                        />
                        <ActionCard
                            title="Library"
                            icon={<BookOpen size={22} color={colors.secondary} />}
                            color={colors.secondary}
                            description="Digital Assets"
                            onPress={() => onNavigate('resources')}
                            colors={colors}
                            shadowStyle={shadows.sm}
                        />
                    </View>
                    <View style={styles.gridRow}>
                        <ActionCard
                            title="Social"
                            icon={<MessageSquare size={22} color={colors.accent} />}
                            color={colors.accent}
                            description="Campus Feed"
                            colors={colors}
                            shadowStyle={shadows.sm}
                        />
                        <ActionCard
                            title="Transport"
                            icon={<Bus size={22} color="#D97706" />}
                            color="#D97706"
                            description="Live Tracking"
                            colors={colors}
                            shadowStyle={shadows.sm}
                        />
                    </View>
                </View>

                <Text style={[styles.sectionTitle, { color: colors.text.primary }]}>Academic Status</Text>
                <View style={[styles.updateCard, { backgroundColor: colors.surface }, shadows.sm]}>
                    <View style={[styles.updateIconBox, { backgroundColor: isDarkMode ? colors.background : '#FFFBEB' }]}>
                        <Bell size={20} color={isDarkMode ? colors.primary : "#92400E"} />
                    </View>
                    <View style={styles.updateInfo}>
                        <Text style={[styles.updateText, { color: colors.text.primary }]}>Midterm schedules are now available for Spring 2026.</Text>
                        <Text style={[styles.updateTime, { color: colors.text.light }]}>Posted 2h ago</Text>
                    </View>
                </View>

                <View style={{ height: 120 }} />
            </ScrollView>

            {/* Bottom Tab Bar */}
            <View style={styles.tabBarContainer}>
                <View style={[styles.tabBar, { backgroundColor: colors.surface }, shadows.lg]}>
                    <TouchableOpacity style={styles.tabItem}>
                        <Home size={24} color={colors.primary} />
                        <View style={[styles.activeDot, { backgroundColor: colors.primary }]} />
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.tabItem} onPress={() => onNavigate('resources')}>
                        <BookOpen size={24} color={colors.text.light} />
                    </TouchableOpacity>

                    <View style={styles.centerTab}>
                        <TouchableOpacity style={[styles.centerTabButton, { backgroundColor: colors.secondary, borderColor: colors.background }]} activeOpacity={0.8}>
                            <Plus size={32} color={isDarkMode ? colors.background : "white"} />
                        </TouchableOpacity>
                    </View>

                    <TouchableOpacity style={styles.tabItem} onPress={() => onNavigate('gpa')}>
                        <BarChart3 size={24} color={colors.text.light} />
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.tabItem} onPress={onLogout}>
                        <User size={24} color={colors.text.light} />
                    </TouchableOpacity>
                </View>
            </View>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    scrollContent: {
        // padding set dynamically
    },
    header: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginBottom: 30,
        marginTop: 20,
    },
    headerRight: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 15,
    },
    iconButton: {
        width: 44,
        height: 44,
        borderRadius: 14,
        alignItems: 'center',
        justifyContent: 'center',
    },
    greeting: {
        fontSize: 14,
        fontWeight: '600',
    },
    userName: {
        fontSize: 28,
        fontWeight: '900',
        letterSpacing: -0.5,
    },
    profileAvatar: {
        width: 48,
        height: 48,
        borderRadius: 16,
        alignItems: 'center',
        justifyContent: 'center',
        borderWidth: 3,
        borderColor: 'white',
    },
    avatarText: {
        color: 'white',
        fontWeight: '900',
        fontSize: 16,
    },
    banner: {
        borderRadius: 35,
        padding: 30,
        marginBottom: 35,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        overflow: 'hidden',
    },
    bannerLeft: {
        flex: 1,
        zIndex: 2,
    },
    bannerLabel: {
        fontSize: 12,
        fontWeight: '600',
    },
    bannerValue: {
        color: 'white',
        fontSize: 50,
        fontWeight: '900',
        marginVertical: 4,
    },
    standingBadge: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 6,
        backgroundColor: 'rgba(255,255,255,0.15)',
        paddingHorizontal: 12,
        paddingVertical: 6,
        borderRadius: 12,
        alignSelf: 'flex-start',
        marginBottom: 10,
    },
    standingText: {
        color: 'white',
        fontSize: 10,
        fontWeight: '900',
        letterSpacing: 1,
    },
    bannerRight: {
        zIndex: 2,
    },
    progressCircle: {
        width: 70,
        height: 70,
        borderRadius: 35,
        borderWidth: 6,
        borderColor: 'rgba(255,255,255,0.1)',
        alignItems: 'center',
        justifyContent: 'center',
    },
    progressText: {
        color: 'white',
        fontSize: 18,
        fontWeight: '900',
    },
    sectionHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginBottom: 20,
    },
    sectionTitle: {
        fontSize: 20,
        fontWeight: '900',
    },
    viewAllText: {
        fontSize: 14,
        fontWeight: 'bold',
    },
    grid: {
        gap: 15,
        marginBottom: 35,
    },
    gridRow: {
        flexDirection: 'row',
        gap: 15,
    },
    card: {
        flex: 1,
        borderRadius: 24,
        padding: 20,
    },
    iconContainer: {
        width: 48,
        height: 48,
        borderRadius: 16,
        alignItems: 'center',
        justifyContent: 'center',
        marginBottom: 15,
    },
    cardTitle: {
        fontSize: 16,
        fontWeight: '800',
    },
    cardDescription: {
        fontSize: 11,
        marginTop: 4,
        fontWeight: '500',
    },
    tabBarContainer: {
        position: 'absolute',
        bottom: Platform.OS === 'ios' ? 35 : 20,
        left: 20,
        right: 20,
    },
    tabBar: {
        height: 85,
        borderRadius: 30,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-around',
        paddingHorizontal: 15,
    },
    tabItem: {
        alignItems: 'center',
        justifyContent: 'center',
        flex: 1,
        height: '100%',
    },
    activeDot: {
        width: 4,
        height: 4,
        borderRadius: 2,
        marginTop: 6,
    },
    centerTab: {
        width: 60,
        alignItems: 'center',
        justifyContent: 'center',
    },
    centerTabButton: {
        width: 64,
        height: 64,
        borderRadius: 22,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: -55,
        borderWidth: 5,
    },
    updateCard: {
        padding: 20,
        borderRadius: 24,
        flexDirection: 'row',
        alignItems: 'center',
        gap: 15,
    },
    updateIconBox: {
        width: 44,
        height: 44,
        borderRadius: 14,
        alignItems: 'center',
        justifyContent: 'center',
    },
    updateInfo: {
        flex: 1,
    },
    updateText: {
        fontSize: 14,
        lineHeight: 20,
        fontWeight: '500',
    },
    updateTime: {
        fontSize: 11,
        marginTop: 6,
        fontWeight: 'bold',
    }
});
